src/rewrite.ml

```diff
@@ -5,27 +5,145 @@ open Terms
 open Print
 open Console
 open Proofs
+open Solve
 
 (** Logging function for the rewrite tactic. *)
 let log_rewr = new_logger 'w' "rewr" "informations for the rewrite tactic"
 let log_rewr = log_rewr.logger
 
-(** For now, a pattern is just a term. *)
-type pattern = term
+(** A substitution is a mapping from metavariables to terms. *)
+type substitution = (term * term) list
```

Comment:
You could ensure by the type that the first argument is a metavariable. Anyway, I'm not sure if this is the right representation for a substitution. Using Bindlib variables would probably be better than using metavariables. You don't have to fix that now, and we can discuss that later.